src/main/java/org/broadinstitute/hellbender/utils/read/ReadUtils.java

```diff
@@ -74,6 +156,62 @@ public static String prettyPrintSequenceRecords ( SAMSequenceDictionary sequence
         RIGHT_TAIL
     }
 
+    public static final int SAM_READ_PAIRED_FLAG = 0x1;
+    public static final int SAM_PROPER_PAIR_FLAG = 0x2;
+    public static final int SAM_READ_UNMAPPED_FLAG = 0x4;
+    public static final int SAM_MATE_UNMAPPED_FLAG = 0x8;
+    public static final int SAM_READ_STRAND_FLAG = 0x10;
+    public static final int SAM_MATE_STRAND_FLAG = 0x20;
+    public static final int SAM_FIRST_OF_PAIR_FLAG = 0x40;
+    public static final int SAM_SECOND_OF_PAIR_FLAG = 0x80;
+    public static final int SAM_NOT_PRIMARY_ALIGNMENT_FLAG = 0x100;
+    public static final int SAM_READ_FAILS_VENDOR_QUALITY_CHECK_FLAG = 0x200;
+    public static final int SAM_DUPLICATE_READ_FLAG = 0x400;
+    public static final int SAM_SUPPLEMENTARY_ALIGNMENT_FLAG = 0x800;
+
+    public static int getSAMFlagsForRead( final GATKRead read ) {
+        int samFlags = 0;
+
+        if ( read.isPaired() ) {
+            samFlags |= SAM_READ_PAIRED_FLAG;
+        }
+        if ( read.isProperlyPaired() ) {
+            samFlags |= SAM_PROPER_PAIR_FLAG;
+        }
+        if ( read.isUnmapped() ) {
+            samFlags |= SAM_READ_UNMAPPED_FLAG;
+        }
+        if ( read.mateIsUnmapped() ) {
+            samFlags |= SAM_MATE_UNMAPPED_FLAG;
+        }
+        if ( read.isReverseStrand() ) {
+            samFlags |= SAM_READ_STRAND_FLAG;
```

Comment:
this code suggests that the constant should be renamed to `SAM_READ_REV_STRAND_FLAG` or something